plugins/repository-hdfs/src/main/plugin-metadata/plugin-security.policy

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+grant {
+  // used for DomainCombiner
+  permission java.security.SecurityPermission "createAccessControlContext";
+  
+  // set TCCL used for bootstrapping Hadoop Configuration and JAAS
+  permission java.lang.RuntimePermission "setContextClassLoader";
+  
+  // Hadoop UGI
+  permission java.lang.reflect.ReflectPermission "suppressAccessChecks";
+  
+  // UGI triggers JAAS
+  permission javax.security.auth.AuthPermission "getSubject";
+  
+  // JAAS libraries are not loaded with the proper context in Hadoop, hence why the permission is needed here
+  permission java.lang.RuntimePermission "loadLibrary.jaas_nt";
+  
```

Comment:
is there any way to elaborate on what is happening here? I don't understand the problem, it smells like something in hadoop is really broken here, e.g. jar-helling with the JDK itself? (which we don't check for in jarhell). 

Because jaas should be loaded via extensions mechanism (e.g. its in the jdk itself) which has full privileges.